### ISSUE TITLE
Metalog bleeds onto tty1

### ISSUE THREAD

**Issue by shahab-vahedi** [void-linux/void-packages#45653]
### Is this a new report?

Yes

### System Info

Void 6.3.13_1 x86_64 GenuineIntel uptodate hold rrrmdDDFFFFFF

### Package(s) Affected

metalog-20230719_1

### Expected behaviour

TTY1 console should remain intact from the system level logs.

Version `20220214` is fine. I noticed this behavior on `20230719`. I never tried version `20230707` though.

### Actual behaviour

Below is a sample output of my default virtual console (`agetty-tty1`). What I do is:

1. Login to the system which results in logs being shown on the output.
2. doing a user level command (`ls`) which goes OK.
3. Repeating step 2 command with privilege (`doas`) which results in system level logging that ends up in the console.
4. Disabling the `metalog` service.
5. Repeating step 3 without any pollution. 

```
Login: edie
Password:

Aug 18 09:55:11 [login] pam_unix(login:session): session opened for user edie(uid=1000) by edie(uid=0)
Aug 18 09:55:11 [login] pam_unix(login:session): session opened for user edie(uid=1000) by edie(uid=0)
Aug 18 09:55:11 [kernel] [ 2540.865203] elogind-daemon[1779]: New session 4 of user edie.
Last login: Fri Aug 18 09:46:38 on tty1
Aug 18 09:55:11 [login] LOGIN ON tty1 BY edie
Aug 18 09:55:11 [login] LOGIN ON tty1 BY edie

$ ls
a  b

$ doas ls
doas (edie@laptop) password:
Aug 18 09:58:27 [doas] pam_unix(doas:session): session opened for user root(uid=0) by edie(uid=1000)
Aug 18 09:58:27 [doas] edie ran command ls as root from /home/edie/test
a  b
Aug 18 09:58:27 [doas] pam_unix(doas:session): session closed for user root

$ doas sv stop metalog
Aug 18 09:58:55 [doas] pam_unix(doas:session): session opened for user root(uid=0) by edie(uid=1000)
Aug 18 09:58:55 [doas] edie ran command sv stop metalog as root from /home/edie/test
ok: down: metalog: 0s, normally up

$ doas ls
a  b
```

### Steps to reproduce

1. Have the service enabled.
2. Drop into your default virtual console (agetty-tty1).
3. You're going to immediately see all the system logs polluted into the console's output.

**Comment by shahab-vahedi** [comment]
@jovanlanik Could this be related to your earlier [comment](https://github.com/void-linux/void-packages/pull/44912#issuecomment-1625625388)?

**Comment by sgn** [comment]
Seems like the correct action is drop `-v` from the service.

**Comment by jovanlanik** [comment]
Yes, I don't see a reason to leave it as default. According to metalog it's for debugging:
```
-v, --verbose
              Run in verbose mode (useful for debugging behavior).
```

**Comment by shahab-vahedi** [comment]
I've removed the `-v` from my setup and the problem still persists.

```
--- /etc/sv/metalog/run.orig    2023-08-18 17:00:30.902350525 +0200
+++ /etc/sv/metalog/run         2023-08-18 17:01:13.284474084 +0200
@@ -1,4 +1,4 @@
 #!/bin/sh
 exec 1>&2
-[ -r conf ] && . ./conf
-exec metalog ${OPTS=-v}
+#[ -r conf ] && . ./conf
+exec metalog
```

**Comment by shahab-vahedi** [comment]
I found the culprit: The `exec 2>&1` added by this [commit](https://github.com/void-linux/void-packages/commit/351646d5bd739c5767561058967af455834ffc3b). I don't know what the rationale behind that commit was, but undoing it, fixes the problem for me.

```
--- /etc/sv/metalog/run.orig
+++ /etc/sv/metalog/run
@@ -1,5 +1,4 @@
 #!/bin/sh
-exec 2>&1
 exec 1>&2
 [ -r conf ] && . ./conf
 exec metalog ${OPTS=-v}
```

**Comment by aryandb** [comment]
I added `-B` option to daemonized it and my problem did go away.

**Comment by gc-user** [comment]
Thanks for finding the cause (and the solution)!
And yes, that error was there in 20230707, too. I was hoping at the time that the 20230719 was the fix for that bug, but it wasn't so I went back to 20220214.

I didn't report the bug, because I thought it was just me / my system / some "wrong setting" I made - and also, because it would have been a bit hard for me to describe the phenomenon in a way a dev would understand what I mean... :-)

I edited the file as per your post and will see during the next boot if it works for me too, but I'll assume it will.

**Comment by shahab-vahedi** [comment]
> I edited the file as per your post and will see during the next boot if it works for me too, but I'll assume it will.

Now the fix is [merged](https://github.com/void-linux/void-packages/commit/b55cfee29117803250d48746b9f71bf5dc13687d). The new package `metalog-20230719_2` should work fine.

**Comment by gc-user** [comment]
Thanks. I noticed the update among others during the day. Haven't since rebooted, but I'm sure it's fine, just as the workaround worked fine.
